.github/workflows/scripts/register_buildpack/index.js

```diff
@@ -0,0 +1,244 @@
+
+    let orgIDs = [];
+    try {
+        const orgs = await github.orgs.listForUser({
+            username: context.payload.sender.login
+        });
+        orgIDs = orgs.data.map(org => org.id);
+
+    } catch (error) {
+        throw error
+    }
+
+    const filteredOwners = owners.filter(owner => (owner.type === 'github_org' && orgIDs.includes(owner.id)));
+    if (filteredOwners.length > 0) {
+        console.error('user successfully authenticated via github_org')
+        return true
+    }
+
+    return false
+}
+
+async function indexRegistryForBuildpack({github, context}, buildpackInfo, owner, repo) {
+    const nameLength = buildpackInfo.name.length
+
+    let indexPath = ''
+    if (nameLength === 1) {
+        indexPath = Path.join(indexPath, '1')
+    } else if (nameLength === 2) {
+        indexPath = Path.join(indexPath, '2')
+    } else if (nameLength === 3) {
+        indexPath = Path.join(indexPath, '3', buildpackInfo.name.slice(0, 2))
+    } else if (nameLength > 3) {
+        indexPath = Path.join(indexPath, buildpackInfo.name.slice(0, 2), buildpackInfo.name.slice(2, 4))
+    } else {
+        throw new Error(`buildpack name cannot be empty`)
+    }
+
+    indexPath = Path.join(indexPath, `${buildpackInfo.ns}_${buildpackInfo.name}`)
+    try {
+        const {data} = await github.repos.getContent({
+            owner,
+            path: indexPath,
+            repo
+        })
+
+        let buff = new Buffer.from(data.content, 'base64')
+        let fileContent = buff.toString('utf-8').trimEnd()
+        fileContent = fileContent + "\n" + JSON.stringify(buildpackInfo)
+
+        await createOrUpdateFileContents(
+            indexPath,
+            context.payload.issue.title,
+            toBase64(fileContent),
+            github,
+            owner,
+            repo,
+            data.sha
+        )
+
+    } catch (error) {
+        if (error.status === 404) {
+            await createOrUpdateFileContents(
+                indexPath,
+                context.payload.issue.title,
+                toBase64(JSON.stringify(buildpackInfo)),
+                github,
+                owner,
+                repo
+            )
+        } else {
+            throw error
+        }
+    }
+}
+
+async function closeIssue({github, context}, owner, repo, name, color) {
+    try {
+        await github.issues.createLabel({
+            owner,
+            repo,
+            name,
+            color
+        });
+    } catch (error) {
+        // Throw an error if we encounter an error other than "already_exists"
+        if (!error.errors.find(err => err.code === 'already_exists')) {
+            throw error
+        }
+    }
+    await github.issues.addLabels({
+        owner,
+        repo,
+        issue_number: context.payload.issue.number,
+        labels: [name]
+    });
+    await github.issues.update({
+        owner,
+        repo,
+        issue_number: context.payload.issue.number,
+        state: 'closed'
+    })
```

Comment:
I think that these two calls can be combined into one - you can also add labels in an update call. 
ref: https://docs.github.com/en/rest/reference/issues#update-an-issue

Reply:
Nice catch! @dfreilich